api/src/test/java/org/openmrs/notification/db/hibernate/HibernateAlertDAOTest.java

```diff
@@ -9,34 +9,105 @@
 import org.junit.Test;
 import org.openmrs.api.context.Context;
+import org.openmrs.notification.Alert;
 import org.openmrs.notification.AlertService;
+import org.openmrs.notification.db.AlertDAO;
 import org.openmrs.test.BaseContextSensitiveTest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
- * This class tests the hibernate alert data access. TODO Consider changing this and all subsequent
- * tests to use dbunit
+ * This class tests the hibernate alert data access. TODO Consider changing this
+ * and all subsequent tests to use dbunit
  */
 public class HibernateAlertDAOTest extends BaseContextSensitiveTest {
-	
+
+	private static final Logger log = LoggerFactory.getLogger(HibernateAlertDAOTest.class);
+	private HibernateAlertDAO hibernateAlertDAO;
+	private Object getAllAlerts;
+	private Object alertId;
+
 	@Before
 	public void runBeforeEachTest() {
+
+		if (hibernateAlertDAO == null)
+			hibernateAlertDAO = (HibernateAlertDAO) applicationContext.getBean("HibernateAlertDao");
 		authenticate();
 	}
-	
+
 	/**
 	 * Test that you can get alerts
 	 * 
 	 * @throws Exception
 	 */
+	// @Test
+	// public void shouldGetAlerts() {
+
+	// }
+    @Test
+	public void getAllAlerts_shouldReturnTrueIfRequested() {
+		AlertService alertService = Context.getAlertService();
+		if (getAllAlerts != null) {
+			Assert.assertEquals(" ", alertService.getAllAlerts(), alertService.getClass());
+		}
+	}
+    @Test
+	public void getAlert_shouldGetAlertById(){
+		assertEquals(1, AlertService.getAlertId();
+	}
 	@Test
-	public void shouldGetAlerts() {
-		
-		AlertService as = Context.getAlertService();
-		//System.out.println(as.getAllAlerts());
-		
+	public void saveAlert_shouldReturnSavedAlertObjectInDatabase() {
+		Session session = null;
+		Transaction transaction =null;
+		  try {
+		      session = HibernateUtil.getSessionFactory().openSession();
+		      transaction = session.beginTransaction();
+		      transaction.begin();
+		      Alert alert = new Alert();
+		      alert.setAlertsByRole("Alert role meesage",alertRole.getMessage());
+		      alert.setAlertByUuid(12223, alertService.getAlertByUuid);
+		      alert.setAlertByText("",alertService.getText());
+		      session.save(alert);
+		      Assert.assertEquals(alert.setAlertByRole("alert role message"),alert.getRole());
+		      Assert.assertEquals(alert.setAlertByUuid(12344),alertService.getAlertUuid());
+		      Assert.assertEquals(alert.setAlertByText(""),alertService.getText());
+		      Assert.assertEquals(alertService.getSavedAlerts().list(),alertService.getClass());
+		      session.commit();
+		      
+	} catch (Exception e) {
+	      e.printStackTrace();
+	    } finally {
+	      if (session != null) {
+	        session.close();
+	      }
+	    }
+
+	    HibernateUtil.shutdown();
+	    
+	 }
+	 @Test
+	public void deleteAlert_shouldDeleteSavedAlertsFromDatabaseIfRequested() {
+	 	Object alertDAO;
+		AlertService.setAlertDAO(alertDAO);
+	    Alert alert = new Alert((int) 1L);
```

Comment:
Can't we just pass 1 instead of (int) 1L ?

Reply:
sure i think passing 1 as integer makes sense instead of passing 1L, am going to make new changes

Reply:
unfortunately am failing to squash my commits, cc.@gcliff, @tendomart ,@varung
https://wiki.openmrs.org/display/docs/Git-Squashing+commits?preview=/36672180/216367785/git_log.PNG
am following this wiki page, some info here thanks

Reply:
Currently am not working on sqaushing not because am failing but because the first branch was deleted unexpectedly. unfortunately @herbert24 is working on it